src/d/d_wood.cpp

```diff
@@ -4,184 +4,899 @@
+  mDoMtx_stack_c::now[1][3] = mDoMtx_stack_c::now[1][3] + (mPos).y;
+  mDoMtx_stack_c::now[2][3] = mDoMtx_stack_c::now[2][3] + (mPos).z;
+  mDoMtx_concat(j3dSys.getViewMtx(), mDoMtx_stack_c::now, mModelViewMtx);
+
+  mDoMtx_copy(anim[iVar1].mTrunkModelMtx, mDoMtx_stack_c::get());
+  mDoMtx_stack_c::now[0][3] = (mPos).x;
+  mDoMtx_stack_c::now[1][3] = (mPos).y;
+  mDoMtx_stack_c::now[2][3] = (mPos).z;
+  mDoMtx_concat(j3dSys.getViewMtx(), mDoMtx_stack_c::get(),
+                mTrunkModelViewMtx);
 
-/* 800BEE9C-800BEEA0       .text cc_hit_after_cut__Q25dWood6Unit_cFPQ25dWood8Packet_c */
-void dWood::Unit_c::cc_hit_after_cut(dWood::Packet_c*) {
-    /* Nonmatching */
+  mDoMtx_concat(j3dSys.getViewMtx(), mShadowModelMtx,
+                mShadowModelViewMtx);
 }
 
+/* 800BEA28-800BEA50       .text clear__Q25dWood6Unit_cFv */
+void dWood::Unit_c::clear() { cLib_memSet(this, 0, 0x18c); }
+
+/* 800BEA50-800BEE9C       .text
+ * cc_hit_before_cut__Q25dWood6Unit_cFPQ25dWood8Packet_c */
+void dWood::Unit_c::cc_hit_before_cut(dWood::Packet_c *packet) {
+  int animIdx;
+  int oldAnimIdx;
+
+  dCcMassS_HitInf inf;
+  fopAc_ac_c *actor;
+  u32 ret = dComIfG_Ccsp()->ChkMass(&mPos, &actor, &inf);
+
+  if (mAnimCooldown > 0) {
+    mAnimCooldown -= 1;
+  }
+
+  // Evaluate for attacks that will not cut us down
+  if ((ret & 1)) {
+    cCcD_Obj *atHitObj = inf.GetAtHitObj();
+    if (atHitObj != NULL &&
+        // @TODO: I think the cCcD_ObjAtType enum is reversed. E.g. the first
+        // one should be Machete (bit 21) not Wind
+        (atHitObj->ChkAtType(AT_TYPE_WIND) ||
+         atHitObj->ChkAtType(AT_TYPE_BOMB) ||
+         atHitObj->ChkAtType(AT_TYPE_FIRE) ||
+         atHitObj->ChkAtType(AT_TYPE_NORMAL_ARROW) ||
+         atHitObj->ChkAtType(AT_TYPE_FIRE_ARROW) ||
+         atHitObj->ChkAtType(AT_TYPE_ICE_ARROW) ||
+         atHitObj->ChkAtType(AT_TYPE_LIGHT_ARROW) ||
+         atHitObj->ChkAtType(AT_TYPE_HOOKSHOT))) {
+
+      // Clear the hit bit so that we don't get cut down
+      ret &= ~0x01;
+
+      if (actor != NULL && mAnimCooldown == 0) {
+        animIdx = packet->search_anm(Anm_c::Mode_PushInto);
+        mAnimCooldown = 20;
+
+        // Play the cut sound
+        JAIZelBasic::getInterface()->seStart(JA_SE_OBJ_TREE_SWING, &mPos,
+                                             0, 0, 1.0, 1.0, -1.0, -1.0, 0);
+
+        // If we are currently performing a basic animation, assign a new
+        // animation
+        oldAnimIdx = mAnmIdx;
+        if ((oldAnimIdx < 8) && (animIdx != -1)) {
+          mAnmIdx = animIdx;
+        }
+
+        // If we were able to allocate an animation (or we already have one),
+        // start the "PushInto" (shrinking) animation
+        if ((mAnmIdx >= 8) &&
+            packet->get_anm(mAnmIdx)->get_mode() >=
+                Anm_c::Mode_PushInto) {
+          s16 targetAngle =
+              cLib_targetAngleY(&(actor->current).pos, &mPos);
```

Comment:
```suggestion
              cLib_targetAngleY(&actor->current.pos, &mPos);
```
parentheses aren't necessary here. same for the other 15 places in this file if you ctrl+f for `).`